src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/RunnableProjectGenerator.cs

```diff
@@ -118,58 +118,41 @@ public IParameterSet GetParametersForTemplate(IEngineEnvironmentSettings environ
-
-            Regex localeFileRegex = new Regex(
-                @"
-                ^
-                (?<locale>
-                    [a-z]{2}
-                    (?:-[A-Z]{2})?
-                )
-                \."
-                + Regex.Escape(TemplateConfigFileName)
-                + "$",
-                RegexOptions.IgnorePatternWhitespace);
-
             IList<ITemplate> templateList = new List<ITemplate>();
             localizations = new List<ILocalizationLocator>();
 
-            foreach (IFile file in folder.EnumerateFiles("*" + TemplateConfigFileName, SearchOption.AllDirectories))
+            foreach (IFile file in folder.EnumerateFiles(TemplateConfigFileName, SearchOption.AllDirectories))
             {
-                if (string.Equals(file.Name, TemplateConfigFileName, StringComparison.OrdinalIgnoreCase))
-                {
-                    IFile hostConfigFile = file.MountPoint.EnvironmentSettings.SettingsLoader.FindBestHostTemplateConfigFile(file);
-
-                    if (TryGetTemplateFromConfigInfo(file, out ITemplate template, hostTemplateConfigFile: hostConfigFile))
-                    {
-                        templateList.Add(template);
-                    }
-
-                    continue;
-                }
+                IFile hostConfigFile = file.MountPoint.EnvironmentSettings.SettingsLoader.FindBestHostTemplateConfigFile(file);
 
-                Match localeMatch = localeFileRegex.Match(file.Name);
-                if (localeMatch.Success)
+                if (TryGetTemplateFromConfigInfo(file, out ITemplate template, hostTemplateConfigFile: hostConfigFile))
                 {
-                    string locale = localeMatch.Groups["locale"].Value;
+                    templateList.Add(template);
 
-                    if (TryGetLangPackFromFile(file, out ILocalizationModel locModel))
+                    IDirectory localizeFolder = file.Parent.DirectoryInfo("localize");
+                    if (localizeFolder != null && localizeFolder.Exists)
                     {
-                        ILocalizationLocator locator = new LocalizationLocator()
+                        foreach (IFile locFile in localizeFolder.EnumerateFiles("*" + LocalizationFileSuffix, SearchOption.AllDirectories))
                         {
-                            Locale = locale,
-                            MountPointUri = source.MountPointUri,
-                            ConfigPlace = file.FullPath,
-                            Identity = locModel.Identity,
-                            Author = locModel.Author,
-                            Name = locModel.Name,
-                            Description = locModel.Description,
-                            ParameterSymbols = locModel.ParameterSymbols
-                        };
-                        localizations.Add(locator);
+                            string locale = locFile.Name.Substring(0, locFile.Name.Length - LocalizationFileSuffix.Length);
```

Comment:
should we check if it's  a valid locale here? We can have a file called `bla.templatestrings.json`

Reply:
I think we should do that, but I want that to be a part of change where `string Locale`s are replaced with `CultureInfo Locale`s everywhere in the solution. Until we do that, I think this one is harmless.